lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -804,165 +723,102 @@ mkTipSyncClient
-                        >>= handleAcquireFailure
-
-                AnyCardanoEra ShelleyEra -> do
-                    pp <- bracketQuery "GetCurrentPParams" tr $ localStateQueryQ `send`
-                        (CmdQueryLocalState pt (QueryIfCurrentShelley Shelley.GetCurrentPParams))
-                    sequence (handleParamsUpdate fromShelleyPParams <$> mb <*> pp <*> sp)
-                        >>= handleAcquireFailure
-                AnyCardanoEra AllegraEra -> do
-                    pp <- bracketQuery "GetCurrentPParams" tr $ localStateQueryQ `send`
-                        (CmdQueryLocalState pt (QueryIfCurrentAllegra Shelley.GetCurrentPParams))
-                    sequence (handleParamsUpdate fromShelleyPParams <$> mb <*> pp <*> sp)
-                        >>= handleAcquireFailure
-                AnyCardanoEra MaryEra -> do
-                    pp <- bracketQuery "GetCurrentPParams" tr $ localStateQueryQ `send`
-                        (CmdQueryLocalState pt (QueryIfCurrentMary Shelley.GetCurrentPParams))
-                    sequence (handleParamsUpdate fromShelleyPParams <$> mb <*> pp <*> sp)
-                        >>= handleAcquireFailure
-
-        handleAcquireFailure
-            :: Either AcquireFailure ()
-            -> m ()
-        handleAcquireFailure = \case
-            Right () ->
-                pure ()
-            Left e ->
-                traceWith tr $ MsgLocalStateQueryError TipSyncClient $ show e
-
-        handleParamsUpdate
-            :: (Maybe Bound -> p -> W.ProtocolParameters)
-            -> (Maybe Bound)
-            -> (Either (MismatchEraInfo (CardanoEras StandardCrypto)) p)
-            -> (Either (MismatchEraInfo (CardanoEras StandardCrypto)) W.SlottingParameters)
-            -> m ()
-        handleParamsUpdate convert boundM epp esp = case (epp, esp) of
-            (Right pp, Right sp) -> do
-                onPParamsUpdate' (convert boundM pp, sp)
-
-            (Left mismatch, _) ->
-                traceWith tr $ MsgLocalStateQueryEraMismatch mismatch
-
-            (_, Left mismatch) ->
-                traceWith tr $ MsgLocalStateQueryEraMismatch mismatch
-
-        queryInterpreter
-            :: Point (CardanoBlock StandardCrypto)
-            -> m ()
-        queryInterpreter pt = do
-            res <- localStateQueryQ `send` CmdQueryLocalState pt (QueryHardFork GetInterpreter)
-            case res of
-                Left (e :: AcquireFailure) ->
-                    traceWith tr $ MsgLocalStateQueryError TipSyncClient $ show e
-                Right interpreter -> do
-                    traceWith tr $ MsgInterpreter interpreter
-                    onInterpreterUpdate interpreter
-
-        W.GenesisParameters
+    let queryParams = do
+            mb <- currentEra >>= \case
+                AnyCardanoEra ShelleyEra ->
+                    LSQry $ QueryAnytimeShelley GetEraStart
+                _ ->
+                    LSQry $ QueryAnytimeMary GetEraStart
```

Comment:
Hmm. This seems wrong (not that this PR is changing anything but we should look into this).

Reply:
Yes. I know. I'm not sure what it's supposed to do. When we change `hardfork_at` to return all eras should make it right though.